src/mps/mpo.jl

```diff
@@ -258,3 +293,66 @@ function densityMatrixApplyMPO(A::MPO, psi::MPS; kwargs...)::MPS
     psi_out.rlim_ = 2
     return psi_out
 end
+
+function nmultMPO(A::MPO, B::MPO; kwargs...)::MPO
+    cutoff::Float64 = get(kwargs, :cutoff, 1e-14)
+    resp_degen::Bool = get(kwargs, :respect_degenerate, true) 
+    maxdim::Int = get(kwargs,:maxdim,max(maxDim(A), maxDim(B)))
+    mindim::Int = max(get(kwargs,:mindim,1), 1)
+    n = length(A)
+    n != length(B) && throw(DimensionMismatch("lengths of MPOs A ($n) and B ($(length(B))) do not match"))
+    A_ = copy(A)
+    position!(A_, 1)
+    B_ = copy(B)
+    position!(B_, 1)
+
+    links_A = findinds.(A.A_, "Link")
+    links_B = findinds.(B.A_, "Link")
+
+    for i in 1:n
+        if length(commoninds(findinds(A_[i], "Site"), findinds(B_[i], "Site"))) == 2
+            A_[i] = prime(A_[i], "Site")
+        end
+    end
+    res = similar(A_)
+    # otherwise we will end up modifying the elements of A!
+    res.A_ = deepcopy(A_.A_)
+    # need to reindex link indices of res
```

Comment:
Maybe would be good to make this a separate function: seems like something we could use again in the future